src/components/modals/AddRoomModal.tsx

```diff
@@ -173,7 +232,48 @@ export default function AddRoomModal({ setOpen, ...props }: ModalProps) {
                         </div>
                     </>
                     <>
-                        <Submit label="Create" disabled={!canCreate} />
+                        <Submit label={createSubmitLabel} disabled={!canCreate} />
+                    </>
+                </Form>
+            ) : isTokenGatedRoom ? (
+                <Form onSubmit={onSubmitCreate}>
+                    <>
+                        <Label htmlFor="roomName">Room Name</Label>
+
+                        <TextField id="roomName" value={roomName} disabled={true} />
+                        <Label htmlFor="tokenAddress">Token Address</Label>
+
+                        <TextField id="tokenAddress" value={tokenAddress} disabled={true} />
+
+                        <Label htmlFor="tokenStandard">Token Standard</Label>
+
+                        <TextField id="tokenStandard" value={tokenType.standard} disabled={true} />
+
+                        {tokenType.hasIds && (
+                            <>
+                                <Label htmlFor="tokenId">Token ID</Label>
+                                <TextField
+                                    id="tokenId"
+                                    value={tokenId}
+                                    onChange={(e) => void setTokenId(parseInt(e.target.value))}
+                                />
+                            </>
+                        )}
+
+                        {tokenType.isCountable && (
+                            <>
+                                <Label htmlFor="minTokenAmount">Minimum Token Amount</Label>
+                                <TextField
+                                    id="minTokenAmount"
+                                    value={minTokenAmount}
```

Comment:
`minTokenAmount` holds a number and `parseFloat('')` is a `NaN`. If you remove the initial 0 the field will display an unchangeable `NaN`. Idea:
- let's hold and use a string here, and
- hold its numeric value in a separate consts, and
- use that numeric value in your logic (zero when invalid).